File: progressões/soma-pg.py
def somaPG(inicio, razao, termos):
    if n > 0 and razao != 0:
        termo = inicio
        lista_termos = [termo]
        for i in range(1, termos):
            termo *= razao
            lista_elementos.append(termo)
        return sum(lista_termos)

    else:
        return print('Deve haver pelo menos um termo e a razão deve ser diferente de 0')

File: progressões/soma-PG.js
function somaPG(inicio, razao, termos) {
    let somatorio = inicio;
    let array_termos = []
    let reducer = (previousValue, currentValue) => previousValue + currentValue;
    
    if(termos > 1 && razao != 0 && Number.isInteger(termos)){
        for(let i = inicio; i <= (inicio)*(Math.pow(razao, termos-1)); i *= Math.abs(razao)){
            array_termos.push(somatorio)
            somatorio *= razao;
        }
        return array_termos.reduce(reducer);
    }
    else {
        console.log('É necessário haver pelo menos um termo e a razão ser diferente de 0')
    }
}

console.log(somaPG(1,2,5))